docs/programming-oak.md

```diff
@@ -30,58 +30,26 @@ things from there.
-execution begins. However, Node authors don't _have_ to implement this function
-themselves; for a Node which receives messages (a combination of bytes and
-handles) that can be
+An Oak Node needs to provide a single [entrypoint](abi.md#exported-function),
+which is the point at which Node execution begins. However, Node authors don't
+_have_ to implement this function themselves; for a Node which receives messages
+(a combination of bytes and handles) that can be
 [decoded](https://project-oak.github.io/oak/sdk/doc/oak/io/trait.Decodable.html)
 into a Rust type, there are helper functions in the Oak SDK that make this
-easier.
+easier, and are the idiomatic and recommended way of implementing Applications.
 
 To use these helpers, an Oak Node should be a `struct` of some kind to represent
 the internal state of the Node itself (which may be empty), implement the
-[`oak::Node`](https://project-oak.github.io/oak/sdk/doc/oak/trait.Node.html)
+[`oak::CommandHandler`](https://project-oak.github.io/oak/sdk/doc/oak/trait.CommandHandler.html)
 trait for it, then define an
-[`entrypoint`](https://project-oak.github.io/oak/sdk/doc/oak/macro.entrypoint.html)
+[`entrypoint_command_handler!`](https://project-oak.github.io/oak/sdk/doc/oak/macro.entrypoint_command_handler.html)
 so the Oak SDK knows how to instantiate it.
 
-The defined entrypoint should run an `oak::run_event_loop` function that is
-specified with a channel handle (used for reading messages) and an Oak Node that
-receives these messages. For example, such a handle could be provided by the
-gRPC server pseudo-Node:
-
-<!-- prettier-ignore-start -->
-[embedmd]:# (../examples/machine_learning/module/rust/src/lib.rs Rust /^oak::entrypoint!\(grpc_oak_main.*/ /}\);$/)
-```Rust
-oak::entrypoint!(grpc_oak_main<ConfigMap> => |_receiver| {
-    oak::logger::init_default();
-    let node = Node {
-        training_set_size: 1000,
-        test_set_size: 1000,
-        config: None,
-        model: NaiveBayes::new(),
-    };
-    let grpc_channel =
-        oak::grpc::server::init("[::]:8080").expect("could not create gRPC server pseudo-Node");
-    oak::run_command_loop(node, grpc_channel.iter());
-});
-```
-<!-- prettier-ignore-end -->
-
-If the Node needs per-instance state, this Node `struct` is an ideal place to
-store it. For example, the running average example has a `Node` `struct` with a
-running sum and count of samples:
-
-<!-- prettier-ignore-start -->
-[embedmd]:# (../examples/running_average/module/rust/src/lib.rs Rust /.*struct Handler {.*/ /^}/)
-```Rust
-struct Handler {
-    sum: u64,
-    count: u64,
-}
-```
-<!-- prettier-ignore-end -->
+The
+[`entrypoint_command_handler!`](https://project-oak.github.io/oak/sdk/doc/oak/macro.entrypoint_command_handler.html)
+macro defines the exported function using the lower-level
+[`entrypoint!`](https://project-oak.github.io/oak/sdk/doc/oak/macro.entrypoint.html)
+macro (which requires a function to run as the Node entry point).
```

Comment:
nit: elsewhere uses "entrypoint" (no space).

Reply:
Done here and elsewhere.